controller/index.js

```diff
@@ -255,54 +226,48 @@ var Controller = function (agol, BaseController) {
         }
 
         // create the file path
-        path = controller._createFilePath(key, req.params)
+        path = controller._createFilePath(req.key, req.params)
         // the file name for the export
-        fileParams.fileName = controller._createName(info, key, req.params.format)
+        fileParams.fileName = controller._createName(info, req.key, req.params.format)
 
         // does the data export already exist?
         agol.files.exists(path, fileParams.fileName, function (exists, path) {
 
-          // get the item data before we check for
-          agol.find(req.params.id, function (err, data) {
+          // save the item layer
+          req.query.layer = (!parseInt(req.params.layer, 0)) ? 0 : req.params.layer
+
+          agol.getItem(req.portal, req.params.item, req.query, function (err, itemJson) {
             if (err) {
               return res.status(500).send(err)
             }
-            // save the item layer
-            req.query.layer = (!parseInt(req.params.layer, 0)) ? 0 : req.params.layer
-
-            agol.getItem(data.host, req.params.item, req.query, function (err, itemJson) {
-              if (err) {
-                return res.status(500).send(err)
-              }
 
-              if (exists) {
-                return agol.isExpired(info, req.query.layer, function (err, isExpired) {
-                  fileParams.err = err
-                  if (!isExpired) {
-                    return controller._returnFile(req, res, path, fileParams.fileName)
-                  }
-
-                  // if it's expired, then remove the data and request a new file
-                  agol.dropItem(data.host, req.params.item, req.query, function () {
-                    req.query.format = req.params.format
-                    controller._getItemData(req, res, function (err, itemJson) {
-                      fileParams.err = err
-                      fileParams.itemJson = itemJson
-                      fileParams.data = (itemJson && itemJson.data && itemJson.data[0]) ? itemJson.data[0] : null
-                      // var used to request new files if needed.
-                      controller._requestNewFile(fileParams)
-                    })
+            if (exists) {
+              return agol.isExpired(info, req.query.layer, function (err, isExpired) {
+                fileParams.err = err
+                if (!isExpired) {
+                  return controller._returnFile(req, res, path, fileParams.fileName)
+                }
+
+                // if it's expired, then remove the data and request a new file
+                agol.dropItem(req.portal, req.params.item, req.query, function () {
+                  req.query.format = req.params.format
+                  controller._getItemData(req, res, function (err, itemJson) {
+                    fileParams.err = err
```

Comment:
what's going on with the `err` object being attached to fileParams from different places down the line here?